File: nhanes/read.py
import pandas as pd
import os

pa = "/home/kshedden/data/Teaching/nhanes/2017-2018"

fn = ["DEMO_J.csv.gz", "BMX_J.csv.gz", "BPX_J.csv.gz"]

da = []
for f in fn:
    dx = pd.read_csv(os.path.join(pa, f))
    da.append(dx)

df = pd.merge(da[0], da[1], how="left", on="SEQN")
df = pd.merge(df, da[2], how="left", on="SEQN")

df["RIAGENDR"] = df["RIAGENDR"].replace([1, 2], ["M", "F"])
df["RIDRETH1"] = df["RIDRETH1"].replace([1, 2, 3, 4, 5], ["MA", "OH", "NHW", "NHB", "Other"])

df = df.loc[df.RIDAGEYR >= 18, :]

File: nhanes/read.R
library(dplyr)
library(readr)

pa = "/home/kshedden/data/Teaching/nhanes/2017-2018"

fn = c("DEMO_J.csv.gz", "BMX_J.csv.gz", "BPX_J.csv.gz")

df = NULL
for (f in fn) {
    dx = read_csv(sprintf("%s/%s", pa, f))
    if (is.null(df)) {
        df = dx
    } else {
        df = left_join(df, dx, "SEQN")
    }
}

df = df %>% mutate(RIAGENDR=recode(df$RIAGENDR, `1`="M", `2`="F"))
df = df %>% mutate(RIDRETH1=recode(df$RIDRETH1, `1`="MA", `2`="OH", `3`="NHW", `4`="NHB", `5`="Other"))

df = df %>% filter(RIDAGEYR >= 18)
